test/gtest/ucp/test_ucp_am.cc

```diff
@@ -820,6 +820,210 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
 UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
 
 
+class test_ucp_am_nbx_send_copy_header : public test_ucp_am_nbx {
+public:
+    test_ucp_am_nbx_send_copy_header()
+    {
+        modify_config("PROTO_ENABLE", "n");
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_values(variants, test_ucp_am_base::get_test_variants_prereg, 0);
+        add_variant_values(variants, test_ucp_am_base::get_test_variants_prereg,
+                           UCP_AM_SEND_FLAG_REPLY, "reply");
+    }
+
+protected:
+    void check_eager(size_t size, size_t header_size)
+    {
+        mem_buffer sbuf(size, tx_memtype());
+        sbuf.pattern_fill(SEED);
+        m_hdr_copy.resize(header_size);
+        ucs::fill_random(m_hdr_copy);
+        ucs_status_ptr_t pending_sptr;
+        m_hdr                  = m_hdr_copy;
+        req_counter            = 0;
+        cb_counter             = 0;
+        unsigned flags         = UCP_AM_SEND_FLAG_COPY_HEADER;
+        ucp_mem_h memh         = NULL;
+        unsigned data_cb_flags = 0;
+        ucp::data_type_desc_t sdt_desc(m_dt, sbuf.ptr(), size);
+
+        if (prereg()) {
+            memh = sender().mem_map(sbuf.ptr(), size);
+        }
+
+        set_am_data_handler(receiver(), TEST_AM_NBX_ID, am_data_cb, this,
+                            data_cb_flags);
+
+        pending_sptr = fill_sq(sdt_desc, memh, header_size,
+                               get_send_flag() | flags);
+        if (pending_sptr == NULL) {
+            UCS_TEST_MESSAGE << "Failed to get pending request";
+        }
+
+        ucs::fill_random(m_hdr_copy);
+        while (progress());
+        wait_for_value(&cb_counter, req_counter);
+        request_wait(pending_sptr);
+        EXPECT_EQ(req_counter, cb_counter);
+
+        if (prereg()) {
+            sender().mem_unmap(memh);
+        }
+    }
+
+    void check_rndv(size_t size, size_t header_size)
```

Comment:
this function seems very similar to check_eager, can we unite common code?
seems the only difference would be send flags (EAGER/RNDV)